### PR TITLE
Compute: fix playback tests

### PR DESCRIPTION
**Pull request by xccc-msft**

### CODE REVIEW

File: azure-mgmt-compute/src/main/java/com/azure/management/compute/VirtualMachineIdentity.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
  * The VirtualMachineIdentity model.
  */
 @Fluent
-public final class VirtualMachineIdentity {
+public class VirtualMachineIdentity {
```
Comment (yaohaizh) [inline azure-mgmt-compute/src/main/java/com/azure/management/compute/VirtualMachineIdentity.java]: Why change the final statements?

File: azure-mgmt-compute/src/test/java/com/azure/management/compute/VirtualMachineAvailabilityZoneOperationsTests.java [hunk 2]
```diff
@@ -48,16 +50,41 @@ protected void cleanUpResources() {
...
+        ResourceGroup resourceGroup = resourceManager.resourceGroups().define(RG_NAME)
+                .withRegion(REGION)
+                .create();
+        //Create network security group
+        NetworkSecurityGroup nsg = networkManager.networkSecurityGroups().define(nsgName)
+                .withRegion(REGION)
+                .withExistingResourceGroup(resourceGroup)
+                .create();
+        //Create public IP address
+        PublicIPAddress pip = networkManager.publicIPAddresses().define(pipDnsLabel)
```
Comment (yaohaizh) [inline azure-mgmt-compute/src/test/java/com/azure/management/compute/VirtualMachineAvailabilityZoneOperationsTests.java]: Please keep the test case aka. example back compatible. Not sure why so much changes for this example?

File: azure-mgmt-compute/src/main/java/com/azure/management/compute/VirtualMachineIdentity.java (hunk 1 above)
Comment (weidongxu-microsoft) [reply azure-mgmt-compute/src/main/java/com/azure/management/compute/VirtualMachineIdentity.java]: Azure-core bug. This is the workaround until they fix.

File: azure-mgmt-compute/src/test/java/com/azure/management/compute/VirtualMachineAvailabilityZoneOperationsTests.java (hunk 2 above)
Comment (weidongxu-microsoft) [reply azure-mgmt-compute/src/test/java/com/azure/management/compute/VirtualMachineAvailabilityZoneOperationsTests.java]: @xccc-msft You might want to double check with Tanyi. After SdkContext move to manager, and code use this SdkContext for random naming for inline resource creation, at least my test runs good on appservice (which previously got 2 kind of inline, StorageAccount, and AppServicePlan).

**Comment by xccc-msft** [comment]
Close this as NSG change is not mandatory for tests here.

**Pull request closed without merging**